lib/literal/array.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class Literal::ArrayGeneric
+	def initialize(type)
+		@type = type
+	end
+
+	def new(*value)
+		Literal::Array.new(@type, value)
+	end
```

Comment:
This importantly makes a copy of the array since it picks up the splat. This is essential to maintain the integrity of the array, since otherwise it could be mutated elsewhere. This value should never be exposed. Even `to_a` should only expose a duplicate.